carp/pytorch/model/architectures/carp_mlm.py

```diff
@@ -0,0 +1,173 @@
+import torch
+import torch.nn.functional as F
+from torchtyping import TensorType, patch_typeguard
+from typeguard import typechecked
+from dataclasses import dataclass
+from carp.configs import CARPConfig, ModelConfig, TrainConfig
+from carp.pytorch.data.mlm_pipeline import MLMBatchElement
+from carp.pytorch.model.architectures import * 
+from carp.pytorch.model.encoders import get_encoder
+from carp.util import mbTokens, generate_indices
+from typing import List
+
+
+
+# CARP MLM differs from normal CARP since the first epoch will solely use an MLM objective to improve data efficiency. 
+# TODO: The learning rate scheduler needs to account for this, so we need a way to register custom LR schedulers.
+# TODO: We need to make sure it saves a CARP MLM checkpoint after the first epoch so that we can convert it to CARP Cloob or CARP momentum
+
+patch_typeguard()
+
+@typechecked
+@register_architecture
+class CARPMLM(BaseModel):
+    def __init__(self, config: ModelConfig):
+        super().__init__(config)
+        encoder_class = get_encoder(config.encoder_type)
+        self.passage_encoder = encoder_class(
+            config.model_path, config.model_arch
+        )
+        self.review_encoder = encoder_class(
+            config.model_path, config.model_arch
+        )
+        self.latent_dim = config.latent_dim
+        self.pass_projector, self.rev_projector = self._make_projection_layers(config)
+        self.logit_scale = nn.Parameter(
+            torch.ones([], device=config.device)
+            * torch.log(torch.tensor([1 / 0.07], device=config.device))
+        )
+        self.clamp_min = torch.log(torch.tensor([1 / 100], device=config.device))
+        self.clamp_max = torch.log(torch.tensor([100], device=config.device))
+
+        self.mlm_mode = True
+
+    def _embed_data(
+        self,
+        x: MLMBatchElement,
+        encoder,
+        projector,
+    ):
+        if self.mlm_mode:
```

Comment:
Could we instead just pass in `mlm_mode` as an argument to the function instead of maintaining the state in the orchestrator? Factoring out the orchestrator is clean, but it doesn't do much and it makes the logic a little convoluted--I had to go back and forth and check a few different files to understand where this was getting controlled. Just having a function in the training script would be less elegant, but it'd be easier to read/understand. As things get added we'll, in all likelihood, want something along the lines of an "orchestrator", but I don't think things are quite that complicated yet.